### PR TITLE
Drop target prioritization

### LINKED ISSUE #943: Drop target prioritizing covered (not top) element
## Bug Description

Per comment below, the overarching issue here is that drop targets are prioritizing the covered element when two elements are in the same area. We should prioritize the visible (top) element. 

## Expected Behaviour

We should prioritize the visible (top) element for drop targets.

## Steps to Reproduce

1. Add a new page
1. Add a media element as element (not bg)
1. Click to set the media element as fill
1. Drag a new media element to replace the fill in what looks to be the fill drop target
1. See that nothing changes (unless you unfill the first media element, in which case you'll see the second media element being background now).

## Screenshots

fill-droptarget

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## QA Instructions
1. Add two image elements
2. Move these on top of each other
3. Now drag a new image element from the Library to drop target
4. Verify that the top image is being replaced with the new image and not the lower image.

### PR DESCRIPTION
**Pull request by wassgha**
Closes #943

Fixes drop target targeting the lower z-index elements instead of the highest element on the stack.

## Testing instructions

See QA Instructions under https://github.com/google/web-stories-wp/issues/943

### CODE REVIEW

File: assets/src/edit-story/components/dropTargets/provider.js [hunk 1]
```diff
@@ -49,13 +49,17 @@ function DropTargetsProvider({ children }) {
   const getDropTargetFromCursor = useCallback(
     (x, y, ignoreId = null) => {
       const underCursor = document.elementsFromPoint(x, y);
+      const sortedDropTargetIds = (currentPage?.elements || [])
```
Comment (dvoytenko) [inline assets/src/edit-story/components/dropTargets/provider.js]: It looks like `getDropTargetFromCursor` is called a lot? Would it be better to do:

```
const sortedDropTargetIds = useMemo(() => ..., [dropTargets, currentPage?.element]);

const getDropTargetsFromCursor = useCallback( () => ..., [sortedDropTargetIds])
```
?

**Review by dvoytenko — CHANGES_REQUESTED** [review]

**Review by barklund — COMMENTED** [review]
This works excellently!

Could we add a karma test for this? It's a fairly complex setup, but should be doable. Maybe also some base tests about drop targets in general, when you're at it?

**Comment by swissspidy** [comment]
FYI @merapi your call to ping the bot ^

**Comment by wassgha** [comment]
@googlebot I consent.

**Comment by merapi** [comment]
@googlebot I consent.

File: assets/src/edit-story/components/canvas/displayLayer.js [hunk 2]
```diff
@@ -43,7 +43,7 @@ function DisplayLayer() {
   );
 
   return (
-    <Layer pointerEvents="none">
+    <Layer data-testid="DisplayLayer" pointerEvents="none">
```
Comment (merapi) [inline assets/src/edit-story/components/canvas/displayLayer.js:46]: Might be not needed after https://github.com/google/web-stories-wp/pull/2386 will be merged.

**Review by merapi — APPROVED** [review]
Works as expected.

**Review by dvoytenko — APPROVED** [review]

**Comment by barklund** [comment]
Did this pass through QA?

**Pull request merged**